src/foam/java/refinements.js

```diff
@@ -384,9 +384,15 @@ foam.CLASS({
     function createJavaPropertyInfo_(cls) {
       var info = this.SUPER(cls);
       var m = info.getMethod('cast');
-      m.body = 'return ( o instanceof Number ) ?'
-            + '((Number)o).intValue() :'
-            + '(int)o;';
+      m.body =
+      `if ( o instanceof Number ) {
```

Comment:
kind of like this return ( o instance of Number ) ? ((Number) o). intValue() : ( o instanceof String ) ? Integer.valueOf((String) o) : (int)o; ?